app/models/user.rb

```diff
@@ -1,7 +1,10 @@
 class User < ApplicationRecord
-  has_many :posts, foreign_key: 'author_id'
+  has_many :posts, foreign_key: 'user_id'
   has_many :comments
   has_many :likes
+  validates :name, presence: true
```

Comment:
- Also kindly add a length check so users don't enter names that are too long which may cause some inconsistencies in your database. eg ` length: { maximum: 100 }`